### PR TITLE
Update to use .NET 8

### PR DESCRIPTION
**Pull request by JamesNK**
~Work in progress. Don't review.~

* Add .NET 8 target
* In .NET 8 target, use `IHttpRequestTimeoutFeature` to disable timeout middleware for streaming methods
* Remove .NET Core 3 and .NET 5 targets (out of support)
* Add .NET 8 named pipes option to benchmarks
* Use argument helpers
* Update tests to consistently target NET462 for .NET Framework

### CODE REVIEW

File: examples/Frameworker/Client/Program.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 #endregion
 
+using System.Net.Http;
```
Comment (amcasey) [inline examples/Frameworker/Client/Program.cs:19]: Why was this needed?

File: perf/benchmarkapps/GrpcAspNetCoreServer/Program.cs [hunk 2]
```diff
@@ -93,6 +84,24 @@ public static IHostBuilder CreateHostBuilder(string[] args)
...
+                        Console.WriteLine($"Named pipe name: {namedPipeName}");
+                        options.ListenNamedPipe(namedPipeName, listenOptions =>
+                        {
+                            ConfigureListenOptions(listenOptions, config, endPoint);
+                        });
+                    }
+                    else
+                    {
+                        // ListenAnyIP will work with IPv4 and IPv6.
+                        // Chosen over Listen+IPAddress.Loopback, which would have a 2 second delay when
```
Comment (amcasey) [inline perf/benchmarkapps/GrpcAspNetCoreServer/Program.cs:98]: Is this in the aspnetcore docs somewhere?

File: test/Grpc.Net.Client.Tests/AsyncClientStreamingCallTests.cs [hunk 3]
```diff
@@ -396,7 +396,7 @@ public override void Write(byte[] buffer, int offset, int count)
             throw new NotImplementedException();
         }
 
-#if !NET472_OR_GREATER
+#if !NET462_OR_GREATER
```
Comment (amcasey) [inline test/Grpc.Net.Client.Tests/AsyncClientStreamingCallTests.cs:399]: What changed here?

File: test/Grpc.Net.Client.Tests/Balancer/RoundRobinBalancerTests.cs [hunk 4]
```diff
@@ -221,7 +220,7 @@ public async Task HasSubchannels_SubchannelStatusChanges_RefreshResolver()
 
         var transportFactory = new TestSubchannelTransportFactory((s, c) =>
         {
-            logger.LogInformation($"Transport factory returning state: {connectState}");
+            logger!.LogInformation($"Transport factory returning state: {connectState}");
```
Comment (amcasey) [inline test/Grpc.Net.Client.Tests/Balancer/RoundRobinBalancerTests.cs:223]: Sometimes, I declare `ILogger logger = null!;`, rather than updating every use site.

**Review by amcasey — APPROVED** [review]
Seems sensible, though I have no way to know whether it's complete.

**Comment by JamesNK** [comment]
FYI @jtattermusch .NET Core 3 and .NET 5 targets are removed from grpc-dotnet projects. Targets on projects moved from grpc/grpc are unchanged.

File: test/Grpc.Net.Client.Tests/AsyncClientStreamingCallTests.cs (hunk 3 above)
Comment (JamesNK) [reply test/Grpc.Net.Client.Tests/AsyncClientStreamingCallTests.cs:399]: Update to tests to consistently test NET462 for .NET Framework.

File: perf/benchmarkapps/GrpcAspNetCoreServer/Program.cs (hunk 2 above)
Comment (JamesNK) [reply perf/benchmarkapps/GrpcAspNetCoreServer/Program.cs:98]: No. This is just what I observed on my computer a few years ago, and this change fixed it.

File: examples/Frameworker/Client/Program.cs (hunk 1 above)
Comment (JamesNK) [reply examples/Frameworker/Client/Program.cs:19]: I'm not sure. It seems like global namespaces for .NET 4.x were updated to remove System.Net.Http. I didn't investigate closely.
